app/src/main/java/it/niedermann/nextcloud/deck/DeckApplication.java

```diff
@@ -17,23 +18,50 @@
     public static final long NO_BOARD_ID = -1L;
     public static final long NO_STACK_ID = -1L;
 
+    private static String PREF_KEY_THEME;
+
     @Override
     public void onCreate() {
-        setAppTheme(isDarkTheme(getApplicationContext()));
+        PREF_KEY_THEME = getString(R.string.pref_key_dark_theme);
+        setAppTheme(getAppTheme(getApplicationContext()));
         super.onCreate();
     }
 
     // -----------------
     // Day / Night theme
     // -----------------
 
-    public static void setAppTheme(boolean darkTheme) {
-        setDefaultNightMode(darkTheme ? MODE_NIGHT_YES : MODE_NIGHT_NO);
+    public static void setAppTheme(DarkModeSetting setting) {
+        setDefaultNightMode(setting.getModeId());
     }
 
-    public static boolean isDarkTheme(@NonNull Context context) {
+    public static DarkModeSetting getAppTheme(Context context) {
         SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(context);
-        return prefs.getBoolean(context.getString(R.string.pref_key_dark_theme), false);
+        String mode;
+        try {
+            mode = prefs.getString(PREF_KEY_THEME, DarkModeSetting.SYSTEM_DEFAULT.name());
+        } catch (ClassCastException e) {
```

Comment:
This `ClassCastException` should never happen in the Deck app. I saw you already wrote a Migration script which removes the `boolean` preference and adds a `String` preference instead.

Reply:
This exception happens when the app is launched the first time after updating, as this method is called before the migration script can run. Therefore, we need try/catch here...

```
E/AndroidRuntime: FATAL EXCEPTION: main
    Process: it.niedermann.nextcloud.deck.dev, PID: 5160
    java.lang.RuntimeException: Unable to create application it.niedermann.nextcloud.deck.DeckApplication: java.lang.ClassCastException: java.lang.Boolean cannot be cast to java.lang.String
        
```